### PR TITLE
Added support for symbolic file links

### LINKED ISSUE #6995: Proper Symlink Support
### Description

Symlink files and folders are not shown as .symlink files and the properties information is empty. Whereas in the original windows file explorer, it correctly shows the filetype as .symlink and has properties information like what's the target and path etc.
Also drag and drop functionality is not working on symlink files and folders in the files app, whereas it works in original windows file explorer app.

### Steps To Reproduce

1. Create a symlink for a file or folder
2. View the filetype and properties in both files app and windows file explorer app

### Expected behavior

1. Proper filetype should be given
2. Drag and drop support for symlinks should be done

### Files Version

v2.0.34.0

### Windows Version

Windows 11 21H2 22000.346

### Relevant Assets

Picture of symlinks in Files app
files1

Picture of symlinks in File Explorer app (which is expected)
explorer1

Picture of symlink properties in Files app vs File Explorer app

FilesVsExplorer1
FilescsExplorer2

Drag & Drop Functionality not working in Files

https://user-images.githubusercontent.com/92441536/142660214-be706681-ce27-47a5-943e-8e51b82446a0.mp4

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #6995

**Details of Changes**
Add details of changes here.
- Adds basic support for symlinks (the ones generated by e.g. `> mklink dest_file.txt source_file.txt`. Currently treated as Shortcut files.

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: Files/Filesystem/StorageEnumerators/Win32StorageEnumerator.cs [hunk 1]
```diff
@@ -10,6 +10,8 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
```
Comment (BanCrash) [inline Files/Filesystem/StorageEnumerators/Win32StorageEnumerator.cs:14]: Are those two necessary (lines 13-14)? Visual studio says no:

image

**Comment by BanCrash** [comment]
@gave92 if you open the shortcut properties it won't let you use the Ok button:
GIF 24-11-2021 13-34-24

I guess shortcut properties should be only readable since the symlink can't be modified at the actual state. Hopefully that will also fixes the issue.

I mean, those fields should be read-only:
image

**Review by BanCrash — DISMISSED** [review]
LGTM.

**Comment by gave92** [comment]
> I mean, those fields should be read-only

Totally agree, explorer does the same. Just did not want to do UI things in this PR =)

> EDIT: Okay, this seems to be also on main with normal shortcuts, so it's not related to this PR.

Ouch, I'll look into this if I can

**Comment by BanCrash** [comment]
@gave92 now it's not compiling because ItemNameRaw is not defined:
image

BTW, I don't know what pipeline failed since I didn't have any issue while compiling lol.

**Comment by gave92** [comment]
You still need to merge main (it's defined on main)

**Comment by gave92** [comment]
Ok, should be good now.

**Review by BanCrash — APPROVED** [review]

**Pull request merged**